packages/python-google-compute-engine/google_compute_engine/distro_lib/tests/helpers_test.py

```diff
@@ -67,42 +67,88 @@ def testCallDhclient(self, mock_call, mock_exists):
+  @mock.patch('google_compute_engine.distro_lib.helpers.subprocess.check_call')
+  def testCallDhclientIpv6(self, mock_call, mock_exists):
+    mock_logger = mock.Mock()
+    mock_exists.side_effect = [True]
     mock_call.side_effect = [
-        None, None, None, subprocess.CalledProcessError(1, 'Test'),
+        None,
+        None,
+        subprocess.CalledProcessError(1, 'Test'),
+        None,
+        None,
+        subprocess.CalledProcessError(1, 'Test'),
     ]
 
-    helpers.CallDhclientIpv6(['a', 'b'], self.mock_logger, 'test_script')
-    helpers.CallDhclientIpv6(['c', 'd'], self.mock_logger, 'test_script')
-    helpers.CallDhclientIpv6(['e', 'f'], self.mock_logger, None)
-    helpers.CallDhclientIpv6(['g', 'h'], self.mock_logger, None)
+    helpers.CallDhclientIpv6(['a', 'b'], mock_logger, 'test_script')
+    helpers.CallDhclientIpv6(['c', 'd'], mock_logger, None)
+    helpers.CallDhclientIpv6(['e', 'f'], mock_logger, None)
+    helpers.CallDhclientIpv6(['g', 'h'], mock_logger, 'test_script',
+                             release_lease=True)
+    helpers.CallDhclientIpv6(['i', 'j'], mock_logger, None, release_lease=True)
+    helpers.CallDhclientIpv6(['k', 'l'], mock_logger, None, release_lease=True)
 
     expected_calls = [
-        mock.call.logger.info(mock.ANY, ['a', 'b']),
-        mock.call.exists('test_script'),
         mock.call.call(
-            ['timeout', '5', 'dhclient', '-1', '-6', '-v', 'a', 'b']),
-        mock.call.logger.info(mock.ANY, ['c', 'd']),
-        mock.call.exists('test_script'),
+            [
+                'timeout', '5','dhclient', '-sf', 'test_script', '-1', '-6',
+                '-v', 'a', 'b'
+            ]),
         mock.call.call(
-            ['timeout', '5', 'dhclient', '-sf', 'test_script', '-1', '-6',
-             '-v', 'c', 'd']),
-        mock.call.logger.info(mock.ANY, ['e', 'f']),
+            [
+                'timeout', '5', 'dhclient', '-1', '-6', '-v', 'c', 'd',
+            ]),
         mock.call.call(
-            ['timeout', '5', 'dhclient', '-1', '-6', '-v', 'e', 'f']),
-        mock.call.logger.info(mock.ANY, ['g', 'h']),
+            [
+                'timeout', '5', 'dhclient', '-1', '-6', '-v', 'e', 'f',
+             ]),
         mock.call.call(
-            ['timeout', '5', 'dhclient', '-1', '-6', '-v', 'g', 'h']),
-        mock.call.logger.warning(mock.ANY, ['g', 'h']),
+            [
+                'timeout', '5', 'dhclient', '-1', '-6', '-r', '-v', 'g', 'h',
+            ]),
+        mock.call.call(
+            [
+                'timeout', '5', 'dhclient', '-1', '-6', '-r', '-v', 'i', 'j',
+            ]),
+        mock.call.call([
+            'timeout', '5', 'dhclient', '-1', '-6', '-r', '-v', 'k', 'l',
+        ]),
     ]
 
-    self.assertEqual(mocks.mock_calls, expected_calls)
+    self.assertEqual(mock_call.mock_calls, expected_calls)
+    mock_logger.assert_has_calls(
+        [
+            mock.call.warning(mock.ANY, ['e', 'f'])
```

Comment:
Trailing brace (though I suspect you can move the mock onto the same line as the bracket).

Reply:
I don't see an update for this?